pyogrio/_io.pyx

```diff
@@ -47,6 +47,13 @@ FIELD_TYPES = [
     None            # OFTInteger64List, List of 64bit integers, not supported
 ]
 
+FIELD_SUBTYPES = [
```

Comment:
I'd suggest making this into a `dict` to make the keys more explicit rather than index order, and then below you can use `FIELD_SUBTYPES.get(field_subtype)` instead of exception handler.

Arguably, `FIELD_TYPES` could have been a `dict` too, but since that list was supposed to be more exhaustive we didn't handle the case where an entry might not be present.

Reply:
`dict.get` is indeed nicer